api/envoy/extensions/filters/http/oauth2/v3/oauth.proto

```diff
@@ -186,6 +190,9 @@ message OAuth2Config {
   // will still process incoming Refresh Tokens as part of the HMAC if they are there. This is to ensure compatibility while switching this setting on. Future
   // sessions would not set the Refresh Token cookie header.
   bool disable_refresh_token_set_cookie = 20;
+
+  // If set to true, enable PKCE (Proof Key for Code Exchange) for the OAuth2 authorization code flow.
+  bool enable_pkce = 21;
```

Comment:
I prefer to remove this and enable PKCE by default, because the OAuth 2.0 [[RFC6749](https://www.rfc-editor.org/rfc/rfc6749)] server responses are unchanged by the PKCE specification, client implementations of this specification do not need to know if the server has implemented this specification or not and SHOULD send the additional parameters as defined in [Section 4](https://www.rfc-editor.org/rfc/rfc7636#section-4) to all servers.

This is recommend by rfc7636:
   
https://www.rfc-editor.org/rfc/rfc7636#section-5